telemeta/forms/haystack_form.py

```diff
@@ -170,30 +155,69 @@ def list_physical_format():
     physical_format = forms.CharField(required=False, label=(_('physical format')), widget=forms.Select(attrs={'style': 'width:100%'}, choices=list_physical_format()))
     code = forms.CharField(required=False, label=(_('code')), widget=forms.TextInput(attrs={'class': 'form-control', 'type': 'search'}))
 
+    def filterInstru(self, query):
+
+        from telemeta.views.search import BooleanSearchView, Erreur
+
+        try:
+            BooleanSearchView().isCorrectQuery(query)
+        except Erreur:
+            return SQ(instruments__startswith=query)
+
+        operateur = "ET"
+        if isinstance(query, list):
+            queryTerms = query
+        else:
+            queryTerms = query.split()
+        sqTab = []
+        valeur = ""
+        while len(queryTerms) != 0:
+            term = queryTerms.pop(0)
+            if term == "ET" or term == "OU":
```

Comment:
you should explain somewhere how those frenchy operators should be used. Why french ?